src/Features/Core/Portable/GenerateFromMembers/AbstractGenerateFromMembersService.cs

```diff
@@ -72,23 +72,39 @@ protected AbstractGenerateFromMembersCodeRefactoringProvider()
             return null;
         }
 
-        // Can use non const fields and properties with setters in them.
+        protected static bool IsReadableInstanceFieldOrProperty(ISymbol symbol)
+            => !symbol.IsStatic && IsReadableFieldOrProperty(symbol);
+
         protected static bool IsWritableInstanceFieldOrProperty(ISymbol symbol)
-            => !symbol.IsStatic &&
-               (symbol is IFieldSymbol field && IsViableField(field) && !field.IsConst ||
-                symbol is IPropertySymbol property && IsViableProperty(property) && property.IsWritableInConstructor());
+            => !symbol.IsStatic && IsWritableFieldOrProperty(symbol);
 
-        protected static bool IsReadableInstanceFieldOrProperty(ISymbol symbol)
-            => !symbol.IsStatic &&
-               (symbol is IFieldSymbol field && IsViableField(field) ||
-                symbol is IPropertySymbol property && IsViableProperty(property) && !property.IsWriteOnly);
+        private static bool IsReadableFieldOrProperty(ISymbol symbol)
+        {
+            switch (symbol)
+            {
+                case IFieldSymbol field when IsViableField(field): return true;
```

Comment:
can you not just do:  ```case IFieldSymbol field: return IsViableField(field);```?  It seems much simpler than having 'when' *and* a boolean.